src/hssm/distribution_utils/dist.py

```diff
@@ -279,19 +279,24 @@ def rng_fn(
                 )
                 out_shape = sims_out.shape[:-1]
                 replace = rng.binomial(n=1, p=p_outlier, size=out_shape).astype(bool)
-                replace = np.stack([replace, replace], axis=-1)
-                n_draws = np.prod(out_shape)
+                replace_n = int(np.sum(replace, axis=None))
+                if replace_n == 0:
+                    return sims_out
+                replace_shape = (*out_shape[:-1], replace_n)
+                replace_mask = np.stack([replace, replace], axis=-1)
+                n_draws = np.prod(replace_shape)
                 lapse_rt = pm.draw(
                     get_distribution_from_prior(cls._lapse).dist(**cls._lapse.args),
                     n_draws,
                     random_seed=rng,
-                ).reshape(out_shape)
-                lapse_response = rng.binomial(n=1, p=0.5, size=out_shape)
+                ).reshape(replace_shape)
+                lapse_response = rng.binomial(n=1, p=0.5, size=replace_shape)
```

Comment:
here I would go with np.random.choice(n_choices, ...) to accomodate models that have more than 2 choices.

if n_choices == 2, we change 0 --> -1, otherwise keep choices as they are.

Reply:
Can we tab this for now since it might take a bit more engineering to figure out where that `n_choices` comes from. `ssms` would need that to figure out how many responses to generate right?

Reply:
yeah. That should be part of model configs actually so should be easily accessible. (can double check but I think it is for ssm-simulators?)

Reply:
Also outputs from ssms.simulator() have the 'metadata' key, which also has that.

Reply:
I mean we do have one step to figure out number of responses from the data, but we might want the user to be a bit more explicit in non-binary cases. In any case we might need a separate PR to address this